prebid/src/main/java/com/uid2/prebid/UID2Prebid.kt

```diff
@@ -0,0 +1,132 @@
+package com.uid2.prebid
+
+import com.uid2.UID2Manager
+import com.uid2.UID2ManagerState
+import com.uid2.UID2ManagerState.Established
+import com.uid2.UID2ManagerState.Refreshed
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import org.prebid.mobile.ExternalUserId
+import org.prebid.mobile.PrebidMobile
+
+/**
+ * Interface to wrap access to [PrebidMobile]. This is used to improve testability, rather than having the [UID2Prebid]
+ * access it via static methods.
+ */
+internal fun interface PrebidExternalUserIdInteractor {
+    operator fun invoke(ids: List<ExternalUserId>)
+}
+
+/**
+ * Prebid integration that will observe a given [UID2Manager] instance and update Prebid when a new [ExternalUserId] is
+ * available. After creating the instance, a consumer must explicitly call [initialize] for this instance to start
+ * observing changes.
+ */
+public class UID2Prebid internal constructor(
+    private val manager: UID2Manager,
+    private val externalUserIdFactory: () -> List<ExternalUserId>,
+    private val prebidInteractor: PrebidExternalUserIdInteractor,
+    dispatcher: CoroutineDispatcher,
+) {
+
+    // We redirect to the logger owned by the UID2Manager, as it's been configured correctly.
+    private val logger = manager.logger
+
+    /**
+     * Constructor.
+     *
+     * @param manager The [UID2Manager] instance to be observed.
+     * @param externalUserIdFactory A factory that will allow the consumer to add any other [ExternalUserId]s that should
+     *                              also be included, rather than just a single list containing only UID2's instance.
+     */
+    @JvmOverloads
+    public constructor(
+        manager: UID2Manager,
+        externalUserIdFactory: () -> List<ExternalUserId> = { emptyList() },
```

Comment:
I wasn't sure how many other `ExternalUserId`s might be set (outside of UID2) by a publisher, so have added support for them to inject any additional ones when we detect our own UID2 changes.